src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -242,54 +248,63 @@ private static BasicStudyInfos toBasicInfos(BasicStudyEntity entity) {
     }
 
     Mono<StudyInfos> getCurrentUserStudy(String studyName, String userId, String headerUserId) {
-        Mono<StudyEntity> studyMono = studyRepository.findStudy(userId, studyName);
-        return studyMono.flatMap(study -> {
-            if (study.isPrivate() && !userId.equals(headerUserId)) {
-                return Mono.error(new StudyException(NOT_ALLOWED));
-            } else {
-                return Mono.just(study);
-            }
-        }).map(StudyService::toInfos);
+        return getStudy(studyName, userId)
+                .switchIfEmpty(Mono.error(new StudyException(STUDY_NOT_FOUND)))
+                .flatMap(study -> {
+                    if (study.isPrivate() && !userId.equals(headerUserId)) {
+                        return Mono.error(new StudyException(NOT_ALLOWED));
+                    } else {
+                        return Mono.just(study);
+                    }
+                }).map(StudyService::toInfos);
     }
 
     Mono<StudyEntity> getStudy(String studyName, String userId) {
-        return studyRepository.findStudy(userId, studyName);
+        return studyRepository.findByUserIdAndStudyName(userId, studyName).map(Mono::just).orElseGet(Mono::empty);
     }
 
-    private Mono<BasicStudyEntity> getStudyCreationRequest(String studyName, String userId) {
-        return studyCreationRequestRepository.findStudy(userId, studyName);
+    private Optional<StudyCreationRequestEntity> getStudyCreationRequest(String studyName, String userId) {
+        return studyCreationRequestRepository.findByUserIdAndStudyName(userId, studyName);
```

Comment:
why not use the same as above:
.map(Mono::just).orElseGet(Mono::empty);

This keeps the code more symmetric